mesh/mesh.hpp

```diff
@@ -789,6 +818,7 @@ class Mesh
    int AddVertex(const Vector &coords);
    /// Mark vertex @a i as nonconforming, with parent vertices @a p1 and @a p2.
    void AddVertexParents(int i, int p1, int p2);
+   int AddVertexAtMidPoint(Array<int> list, int dim = 3);
```

Comment:
This should either pass by const reference, or change to `const int *v, int N` to handle the variable length array. Then avoids the allocation of `Array<int>` repeatedly.